src/utils/moonpay.js

```diff
@@ -0,0 +1,39 @@
+import sendJson from 'fetch-send-json'
+import { ACCOUNT_HELPER_URL } from '../utils/wallet'
+
+const MOONPAY_API_URL = process.env.MOONPAY_API_URL || 'https://api.moonpay.com'
+const MOONPAY_API_KEY = process.env.MOONPAY_API_KEY || 'pk_test_wQDTsWBsvUm7cPiz9XowdtNeL5xasP9'
+export const MOONPAY_BUY_URL = `${process.env.MOONPAY_BUY_URL || 'https://buy.moonpay.io?apiKey='}${MOONPAY_API_KEY}`
+
+export const isMoonpayAvailable = async () => {
+    const moonpayGet = (path) => sendJson('GET', `${MOONPAY_API_URL}${path}?apiKey=${MOONPAY_API_KEY}`)
+    const isAllowed = ({ isAllowed, isBuyAllowed }) => isAllowed && isBuyAllowed
+
+    const ipAddressInfo = await moonpayGet('/v4/ip_address')
```

Comment:
@Patrick1904 given you are raising concern about speed – we can try to make stuff load a bit faster like this:

```js
const [ipAddressInfo, countries, currencies] = await Promise.all([
   moonpayGet('/v4/ip_address'),
   moonpayGet('/v3/countries'),
   moonpayGet('/v3/currencies')
])
```